hooks/GNS.js

```diff
@@ -0,0 +1,193 @@
+11. Differential reference station ID, 0000-4095 - Null if Talker ID is GN, Additional GNS messages follow with Reference station ID
+12. Navigational status (added when the IEC61162-1:2010/NMEA 0183 V4.10 option is selected in the NMEA I/O configuration):
+     - S = Safe
+     - C = Caution
+     - U = Unsafe
+     - V = Not valid for navigation
+13. Checksum
+*/
+
+function isEmpty(mixed) {
+  return (
+    (typeof mixed !== 'string' && typeof mixed !== 'number') ||
+    (typeof mixed === 'string' && mixed.trim() === '')
+  )
+}
+
+function indicator(chars, modes) {
+    const systems = ["GPS","GNSS","Galileo","BeiDou","QZSS"]
+    const indications = chars.map(function (val, i) { return `${systems[i]}: ${modes[val]}` })
+    return indications.join(",")
+  }
+
+module.exports = function (input) {
+  const { id, sentence, parts, tags } = input
+
+  const empty = parts.reduce((e, val) => {
+    if (isEmpty(val)) {
+      ++e
+    }
+    return e
+  }, 0)
+
+  if (empty > 4) {
+    return null
+  }
+
+  const time = parts[0].indexOf('.') === -1 ? parts[0] : parts[0].split('.')[0]
+  const timestamp = utils.timestamp(time, moment.tz('UTC').format('DDMMYY'))
+
+  const mode = {
+    "A": "Autonomous (non-differential)",
+    "D": "Differential mode",
+    "E": "Estimated (dead reckoning) Mode",
+    "F": "RTK Float",
+    "M": "Manual Input Mode",
+    "N": "Constellation not in use, or no valid fix",
+    "P": "Precise (no degradation, like Selective Availability, and hires)",
+    "R": "RTK Integer",
+    "S": "Simulator Mode"
+  }
+
+  const state = {
+    "S": "Safe",
+    "C": "Caution",
+    "U": "Unsafe",
+    "V": "Not valid for navigation"
+  }
+
+  const delta = {
+    updates: [
+      {
+        source: tags.source,
+        timestamp: timestamp,
+        values: [
+          {
+            path: 'navigation.position',
+            value: {
+              longitude: utils.coordinate(parts[3], parts[4]),
+              latitude: utils.coordinate(parts[1], parts[2]),
+            },
+          },
+          {
+            path: 'navigation.gnss.methodQuality',
+            value: indicator(parts[5].split(""), mode),
+          },
+
+          {
+            path: 'navigation.gnss.satellites',
+            value: utils.int(parts[6]),
+          },
+
+          {
+            path: 'navigation.gnss.antennaAltitude',
+            value: utils.float(parts[8]),
+          },
+
+          {
+            path: 'navigation.gnss.horizontalDilution',
+            value: utils.float(parts[7]),
+          },
+
+          {
+            path: 'navigation.gnss.geoidalSeparation',
+            value: utils.float(parts[9]),
+          },
+
+          {
+            path: 'navigation.gnss.differentialAge',
+            value: utils.float(parts[10]),
+          },
+
+          {
+            path: 'navigation.gnss.differentialReference',
+            value: Number(parts[11]),
+          },
+          {
+            path: 'navigation.state',
```

Comment:
Do by isValidSignalKVessek you mean the following function:
[https://github.com/SignalK/nmea0183-signalk/blob/6e2602c0570c7121f493a1c6ab9b29f8aa2b29c1/lib/index.js#L127-L133](url)

Because I couldn't find a function with this specific name

Reply:
To be exact: you should see `toFull(delta).should.be.validSignalK` failing when you output paths that are not in the Signal K specification.

Reply:
Ok yes I see it happen. So should I just remove this line from the GNS file located in the test directory? Or is there another place to change in order to disable the validation?

Reply:
Comment it out please